### PR TITLE
Softmax

### PR DESCRIPTION
**Pull request by bendecoste**
implemented in Syft.NN.Functional like PyTorch does
no GPU support, that is coming in the next pr

### CODE REVIEW

**Comment by iamtrask** [comment]
We might need to update our tutorial as a result of this API level change:

https://docs.google.com/document/d/1WRd7gGLFN0Awtf86AICYIHtg3gfFWLBa5wYTthsB3i0/edit
